### PR TITLE
Update Dijkstra's to use set

### PR DESCRIPTION
**Pull request by Takashiidobe**
Some changes reviewing the dijkstra's shortest path algo

ref: https://github.com/msambol/dsa/issues/9

### CODE REVIEW

File: shortest_path/dijkstras.py [hunk 1]
```diff
@@ -18,27 +17,25 @@ def make_graph():
 
 def dijkstras_heap(G, start='A'):
     shortest_paths = {} 
-    visited = {}
+    visited = set()
```
Comment (Takashiidobe) [inline shortest_path/dijkstras.py:21]: This one is the change noted in the issue about representing the set as a dict when a set would do.

File: shortest_path/dijkstras.py [hunk 2]
```diff
@@ -18,27 +17,25 @@ def make_graph():
...
     heap = []
 
-    for node in list(G.keys()):
+    for node in G.keys():
```
Comment (Takashiidobe) [inline shortest_path/dijkstras.py:24]: G.keys() is an iterator, so there's no need to turn it into a list.

File: shortest_path/dijkstras.py [hunk 3]
```diff
@@ -47,7 +44,7 @@ def dijkstras_heap(G, start='A'):
 
 def dijkstras(G, start='A'):
     shortest_paths = {}
-    unvisited = list(G.keys())
+    unvisited = dict.fromkeys(G.keys())
```
Comment (Takashiidobe) [inline shortest_path/dijkstras.py:48]: The code used to be a list, which would then call remove on `min_node` in a loop. There's no guarantee that `min_node` is at the end of the list, so `remove` has to do a seek to find the `min_node` in the list and remove it, and both operations are O(n). 

If it's changed to a dictionary, finding the `min_node` is O(1) and removing it is also O(1).

File: shortest_path/dijkstras.py [hunk 4]
```diff
@@ -85,5 +81,4 @@ def main():
     print(f'Shortest path from {start}: {shortest_paths}')
     print(f'Shortest path from {start} using heap: {shortest_paths_using_heap}')
 
-
```
Comment (msambol) [inline shortest_path/dijkstras.py]: nit: can you remove this newline change?

File: shortest_path/dijkstras.py [hunk 5]
```diff
@@ -1,6 +1,5 @@
 import heapq
 
-
```
Comment (msambol) [inline shortest_path/dijkstras.py]: nit: can you remove this newline change?

**Review by msambol — CHANGES_REQUESTED** [review]
Two nits otherwise looks great.

**Review by msambol — APPROVED** [review]
Thanks, @Takashiidobe!

**Pull request merged**